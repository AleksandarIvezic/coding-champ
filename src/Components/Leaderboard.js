import React, { useState } from 'react';
import { v4 as uuidv4 } from 'uuid';
import mockCategories from '../Library/mockCategories';
import mockScores from '../Library/mockScores';
import '../Styles/Leaderboard.css';

const Leaderboard = () => {
  const [selectedCategory, setSelectedCategory] = useState('');
  const handleChange = (e) => {
    setSelectedCategory(e.target.value);
  };

  return (
    <>
      <form>
        <label htmlFor="submit-category">
          Choose a category
          <select id="categories" name="categories" defaultValue="" onChange={handleChange}>
            <option value="" disabled> </option>
            {mockCategories.map((category) => (
              <option key={`category-${uuidv4()}`} value={category.name}>{category.name}</option>
            ))}
          </select>
        </label>
      </form>
      {selectedCategory
      && <h3>{selectedCategory}</h3>}
      <table className="leaderboard-table">
        <thead>
          <tr>
            <th>#R</th>
            <th>Name</th>
            <th>Score</th>
          </tr>
        </thead>
        <tbody id="leaderboard-data">
          {selectedCategory.length > 0
          && mockScores[selectedCategory].map((score, index) => (
            <tr key={`${selectedCategory}-${uuidv4()}`}>
              <td>{index + 1}</td>
              <td>{Object.keys(score)[0]}</td>
              <td>{score[Object.keys(score)[0]]}</td>
            </tr>
          ))}
        </tbody>
      </table>
    </>
  );
};

export default Leaderboard;
