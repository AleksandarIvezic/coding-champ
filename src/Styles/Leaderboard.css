.leaderboard-table{
  width: 25%;
  margin: 10px auto;
}