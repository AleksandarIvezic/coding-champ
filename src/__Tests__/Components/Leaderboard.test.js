import React from 'react';
import renderer from 'react-test-renderer';
import Leaderboard from '../../Components/Leaderboard';

it('renders correctly', () => {
  const tree = renderer
    .create(<Leaderboard />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
