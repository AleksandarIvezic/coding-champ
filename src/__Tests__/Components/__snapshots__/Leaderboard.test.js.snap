// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <form>
    <label
      htmlFor="submit-category"
    >
      Choose a category
      <select
        defaultValue=""
        id="categories"
        name="categories"
        onChange={[Function]}
      >
        <option
          disabled={true}
          value=""
        >
           
        </option>
        <option
          value="HTML"
        >
          HTML
        </option>
        <option
          value="CSS"
        >
          CSS
        </option>
        <option
          value="JavaScript"
        >
          JavaScript
        </option>
      </select>
    </label>
  </form>,
  "",
  <table
    className="leaderboard-table"
  >
    <thead>
      <tr>
        <th>
          #R
        </th>
        <th>
          Name
        </th>
        <th>
          Score
        </th>
      </tr>
    </thead>
    <tbody
      id="leaderboard-data"
    />
  </table>,
]
`;
